package/kedro_viz/models/utils.py

```diff
@@ -43,3 +44,27 @@ def get_dataset_type(dataset: "AbstractDataset") -> str:
     abbreviated_module_name = ".".join(dataset.__class__.__module__.split(".")[-2:])
     class_name = f"{dataset.__class__.__qualname__}"
     return f"{abbreviated_module_name}.{class_name}"
+
+
+def extract_data_source(
+    node_type: Union[str, None], data_desc: Dict
+) -> Union[Optional[str], Any]:
+    """Get the relevant source logic used in generating this node's data, based
+    on dataset type
+
+    Args:
+        node_type: Datatype for the dataset object as a string
+        data_desc: Dict whose values store useful contextual information about the dataset
+
+    Returns:
+        String to display to user as the node source code
+    """
+    if node_type is None:
+        return None
+
+    format_type = node_type.split(".")[-1]
+
+    if format_type in {"SQLQueryDataSet", "GBQQueryDataSet"}:
+        return data_desc["sql"]
```

Comment:
@MehdiNV  - I'm not entirely confident in this approach, especially after discussing it with @merelcht. It seems this approach may not be robust, as changes to kedro-datasets could potentially break it. I would recommend collaborating with someone from the framework team to explore ways to improve our approach

Reply:
May be we should stop hardcoding strings and rely on something like -
`if data_node.is_sql_node():
     self.code = dataset_description.get("sql")`
    as suggested by @rashidakanchwala 
    
 - Even accessing the sql key is not maintainable but we do similar logic fetching the file_path and other properties from the describe function
 - The better way of accessing any values would be having getters inside the kedro-datasets plugin something like `sqlDataset.getCode` - this way we wont rely on the underlying key name. But for now we can check the node type and get the sql code.

Reply:
@merelcht -- also let us know if you think the above approach makes sense. It's not ideal but the alternative as Ravi mentioned is to have getter functions in kedro-datasets ?

Reply:
I've pushed commits for the first approach in the meantime, have just seen latest comments though so don't think they're viable; will chat with framework tomorrow to see if there's any way forward as I think this issue could be blocked for us for now.